xbmc/commons/InfoScanner.h

```diff
@@ -0,0 +1,43 @@
+/*
+ *      Copyright (C) 2015 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+#include "utils/URIUtils.h"
+#include "filesystem/File.h"
+
+namespace XbmcCommons {
+  class InfoScanner {
+  public:
+    virtual bool DoScan(const std::string& strDirectory) = 0;
```

Comment:
Please re-order the methods to
* constructor (if present)
* destructor
* virtual methods (if present)
* normal methods (if present)

and add some empty lines between the method definitions for better readability.